SB/SB/Program.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using CsvHelper;
+using NLog;
+using NLog.Config;
+using NLog.Fluent;
+using NLog.Targets;
+
+namespace SB
+
+{
+    
+    internal class Program
+    {
+        private static readonly ILogger logger = LogManager.GetCurrentClassLogger();
+        
+        public static void Main(string[] args)
+        {
+            var config = new LoggingConfiguration();
+            var target = new FileTarget { FileName = @"C:\Work\Logs\SupportBank.log", Layout = @"${longdate} ${level} - ${logger}: ${message}" };
+            config.AddTarget("File Logger", target);
+            config.LoggingRules.Add(new LoggingRule("*", LogLevel.Debug, target));
+            LogManager.Configuration = config;
+            logger.Info("Starting program");   
+           
+            
+            var pathToCvs = "C:\\Work\\Bootcamp\\SB\\SB\\DodgyTransactions2015.csv";
+            bool running = true;
+            while (running)
+            {
+                string input;
+                input = Console.ReadLine();
+                string[] split = input.Split(' ');
+                if (split[0] == "exit")
+                {
+                    running = false;
+                }
+                else if (split.Length == 2 && split[0] == "List" && split[1] == "All")
+                {
+                    ListAll(pathToCvs);
+
+                }else if (split.Length == 3 && split[0] == "List")
+                {
+                    StringBuilder sb = new StringBuilder();
+                    sb.Append(split[1] + " " + split[2]);
+                    string name = sb.ToString();
+                    Console.WriteLine("NAME: " + name);
+                    List(pathToCvs,name);
+                }else{
+                    Console.WriteLine("Invalid command! Try again.");
+                }            
+                Console.WriteLine("-----------------------------------");
+
+            }
+            
+            
+           
+        }
+
+        private static void ListAll(string pathToCsv)

```

Comment:
You should probably split the logic for reading & parsing the input, and for executing the commands.

You can:
1. Have a `Bank` class, which has a list of `People`
2. Each `Person` has a List of `Transactions`
3. The Bank implements `AddTransaction`/`ProcessTransacrion`,  `ListAll()` and `List(Person)`
4. Create a `CsvParser` class, that as a method `ParseCsv(Bank, csvPath)`. It would read the input line by line, and call bank.AddTransaction()
5. The main method would instantiate the objects, read user input and call the relevant methods. Currently, with each command you read the csv again, although you shouldn't need to if you have a Bank object with all the info already saved in memory.